src/main/java/org/gridsuite/directory/server/DirectoryService.java

```diff
@@ -123,70 +131,66 @@ private void emitDirectoryChanged(UUID directoryUuid, String elementName, String
     }
 
     /* methods */
-    public Mono<ElementAttributes> createElement(ElementAttributes elementAttributes, UUID parentDirectoryUuid, String userId) {
+    public ElementAttributes createElement(ElementAttributes elementAttributes, UUID parentDirectoryUuid, String userId) {
         if (elementAttributes.getElementName().isBlank()) {
-            return Mono.error(new DirectoryException(NOT_ALLOWED));
+            throw new DirectoryException(NOT_ALLOWED);
         }
 
-        return assertElementNotExist(parentDirectoryUuid, elementAttributes.getElementName(), elementAttributes.getType())
-            .and(assertAccessibleDirectory(parentDirectoryUuid, userId))
-            .then(insertElement(elementAttributes, parentDirectoryUuid))
-            .doOnSuccess(unused -> {
-                var isCurrentElementPrivate = elementAttributes.getType().equals(DIRECTORY) ? elementAttributes.getAccessRights().getIsPrivate() : null;
-                emitDirectoryChanged(
-                        parentDirectoryUuid,
-                        elementAttributes.getElementName(),
-                        userId,
-                        isPrivateForNotification(parentDirectoryUuid, isCurrentElementPrivate),
-                        false,
-                        NotificationType.UPDATE_DIRECTORY);
-            });
-    }
-
-    private Mono<Void> assertElementNotExist(UUID parentDirectoryUuid, String elementName, String type) {
-        return isElementExists(parentDirectoryUuid, elementName, type)
-            .flatMap(exist -> Boolean.TRUE.equals(exist) ? Mono.error(new DirectoryException(NOT_ALLOWED)) : Mono.empty());
+        assertElementNotExist(parentDirectoryUuid, elementAttributes.getElementName(), elementAttributes.getType());
+        assertAccessibleDirectory(parentDirectoryUuid, userId);
+        ElementAttributes result = insertElement(elementAttributes, parentDirectoryUuid);
+        var isCurrentElementPrivate = elementAttributes.getType().equals(DIRECTORY) ? elementAttributes.getAccessRights().getIsPrivate() : null;
+        emitDirectoryChanged(parentDirectoryUuid,
+                             elementAttributes.getElementName(),
+                             userId,
+                             isPrivateForNotification(parentDirectoryUuid, isCurrentElementPrivate),
+                            false,
+                             NotificationType.UPDATE_DIRECTORY);
+        return result;
+    }
+
+    private void assertElementNotExist(UUID parentDirectoryUuid, String elementName, String type) {
+        if (Boolean.TRUE.equals(isElementExists(parentDirectoryUuid, elementName, type))) {
```

Comment:
Sonar check mark it as code smell if we don't use Boolean.TRUE.equals

Reply:
If Sonar says so... :)

Reply:
I've read online it's because isElementExists returns a Boolean object, which can be null.
Why it is configured to return an object instead of a primitive, that I don't know